### PR TITLE
[@testing-library/react] 10 + [@testing-library/dom] TSDoc comments

### PR DESCRIPTION
**Pull request by jednano**
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes.
   - [@testing-library/react@10](https://github.com/testing-library/react-testing-library/releases/tag/v10.0.0)
   - [@testing-library/dom@7](https://github.com/testing-library/dom-testing-library/releases/tag/v7.0.0)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/testing-library__cypress/index.d.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@
 //                 Airat Aminev <https://github.com/airato>
 //                 Simon Jespersen <https://github.com/simjes>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.0
+// TypeScript Version: 3.8
```
Comment (eps1lon) [inline types/testing-library__cypress/index.d.ts]: Why is this necessary?

File: types/testing-library__react/index.d.ts [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
 // Definitions by: Alex Krolick <https://github.com/alexkrolick>
 //                 Kent C Dodds <https://github.com/kentcdodds>
 //                 Sebastian Silbermann <https://github.com/eps1lon>
 //                 Weyert de Boer <https://github.com/weyert>
 //                 Ifiok Jr. <https://github.com/ifiokjr>
 //                 Daniel Afonso <https://github.com/danieljcafonso>
+//                 Jed Mao <https://github.com/jedmao>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.0
+// TypeScript Version: 3.8
```
Comment (eps1lon) [inline types/testing-library__react/index.d.ts]: Could you explain why we need `import type`? As far as I know this is only useful for isolatedModules and babel. We need a stronger argument for leaving older TS versions behind.

**Review by eps1lon — CHANGES_REQUESTED** [review]
/dom is already covered by https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43102

**Comment by jedmao** [comment]
@eps1lon I'll rebase this on master once #43102 is merged so the additional TSDoc comments are added.

**Comment by eps1lon** [comment]
Blocked until https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43102 is merged

File: types/testing-library__react/index.d.ts [hunk 3]
```diff
@@ -46,6 +47,9 @@ export function render<Q extends Queries>(ui: React.ReactElement, options: Rende
 
 /**
  * Unmounts React trees that were mounted with render.
+ * @note As a general reminder, you should _NOT_ need to use `cleanup` at all as this is done
+ * automatically for you (unless you're using `@testing-library/react/pure` in which case you know
+ * what you're doing... keep using cleanup).
  */
 export function cleanup(): void;
```
Comment (eps1lon) [inline types/testing-library__react/index.d.ts]:
```suggestion
export async function cleanup(): void;
```

**Review by artem-malko — APPROVED** [review]

**Comment by eps1lon** [comment]
@jedmao Could you rebase this? Should only address `@testing-library/react` now.

**Comment by jedmao** [comment]
I don't understand the build error 😕

**Comment by eps1lon** [comment]
@jedmao I think you should start from scratch. Many of these fixes are already published on master.

**Comment by jedmao** [comment]
@eps1lon I rebased on master. I'm just going to close this.

**Comment by kylemh** [comment]
I'm on the latest version and not seeing `waitFor` as an exported module 🤔

**Pull request closed without merging**